cmd/coreos-assembler.go

```diff
@@ -63,17 +63,29 @@ func run(argv []string) error {
 		argv = argv[1:]
 	}
 
-	if cmd == "" {
+	if cmd == "" || cmd == "--help" {
 		printUsage()
 		os.Exit(1)
 	}
 
+	// if the COREOS_ASSEMBLER_REMOTE_SESSION environment variable is
+	// set then we "intercept" the command here and redirect it to
+	// `cosa remote-session exec`, which will execute the commands
+	// via `podman --remote` on a remote machine.
```

Comment:
One thing I'm hazy on is how this works with the "context directory"; e.g. is the `cosa init` and the git clone expected to happen remotely?  Or is it local, and then podman syncs it?

Reply:
it should be able to work either way. You can `cosa init` locally and sync it over first or you can set up the remote session and `cosa init` on the remote.

Reply:
I'm working on the pipeline changes and I'll post them up shortly (testing them now). Trying to work out all the kinks first.